### PR TITLE
[Enhancement] Optimize C++ demos

### PR DESCRIPTION
**Pull request by lzhangzz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: demo/csrc/cpp/pose_tracker_params.inc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) OpenMMLab. All rights reserved.
```
Comment (lvhan028) [inline demo/csrc/cpp/pose_tracker_params.inc:1]: why not `pose_tracker_params.h`?

File: demo/csrc/cpp/utils/visualize.h [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+      rect &= cv::Rect({}, img_.size());
+      img_(rect) *= .35f;
+      cv::putText(img_, text, origin + cv::Point2f(0, text_size.height), font_face, font_scale,
+                  cv::Scalar::all(255), thickness, cv::LINE_AA);
+      return text_size.height;
+    }
+
+    static std::string to_text(int label_id, float score) {
+      std::stringstream ss;
+      ss << label_id << ": " << std::fixed << std::setprecision(1) << score * 100;
```
Comment (lvhan028) [inline demo/csrc/cpp/utils/visualize.h:75]: Why `score * 100`?

Comment (lvhan028) [reply demo/csrc/cpp/utils/visualize.h:75]: suggest `std::precision(2)`

File: demo/csrc/cpp/utils/argparse.h [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        for (int j = 0; j < infos_.size(); ++j) {
+          auto& flag = infos_[j];
+          if (key == flag.name) {
+            args[j] = val;
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          std::cout << "error: unknown option: " << argv[i] << std::endl;
```
Comment (lvhan028) [inline demo/csrc/cpp/utils/argparse.h:111]: The unknown option should be `key` since `argv[i]` is `val` because of Line 98.

Comment (lvhan028) [reply demo/csrc/cpp/utils/argparse.h:111]: Should we return false when there is an unknown option?

File: demo/csrc/cpp/restorer.cxx [hunk 4]
```diff
@@ -2,33 +2,39 @@
...
-#include <opencv2/imgproc/imgproc.hpp>
-#include <string>
+#include "opencv2/imgcodecs/imgcodecs.hpp"
+#include "opencv2/imgproc/imgproc.hpp"
+#include "utils/argparse.h"
+
+DEFINE_ARG_string(model, "Super-resolution model path");
+DEFINE_ARG_string(image, "Input image path");
+DEFINE_string(device, "cpu", R"(Device name, e.g. "cpu", "cuda")");
+DEFINE_string(output, "restorer_output.jpg", "Output image path");
```
Comment (lvhan028) [inline demo/csrc/cpp/restorer.cxx:12]: ".bmp" is expected.

File: demo/csrc/cpp/segmentor.cxx [hunk 5]
```diff
@@ -2,75 +2,46 @@
...
 
-  auto result = segmentor.Apply(img);
+  // apply the detector, the result is an array-like class holding a reference to
+  // `mmdeploy_segmentation_t`, will be released automatically on destruction
+  mmdeploy::Segmentor::Result seg = segmentor.Apply(img);
 
-  auto palette = gen_palette(result->classes + 1);
+  // visualize
+  utils::Visualize v;
+  v.set_palette(utils::Palette::get(FLAGS_palette));
```
Comment (lvhan028) [inline demo/csrc/cpp/segmentor.cxx:38]: If a customized palette is provided other than 'coco' or 'cityscapes', it crashes. 
```
Floating point exception (core dumped)
```

File: demo/csrc/cpp/restorer.cxx (hunk 4 above)
Comment (lzhangzz) [reply demo/csrc/cpp/restorer.cxx:12]: Why do you think ".bmp" is expected? BMP is extremely large and everywhere in the rest of the whole project is using ".jpg".

File: demo/csrc/cpp/pose_tracker_params.inc (hunk 1 above)
Comment (lzhangzz) [reply demo/csrc/cpp/pose_tracker_params.inc:1]: It's not a header.

File: demo/csrc/cpp/utils/visualize.h (hunk 2 above)
Comment (lzhangzz) [reply demo/csrc/cpp/utils/visualize.h:75]: no

**Review by lvhan028 — APPROVED** [review]

**Review by irexyc — APPROVED** [review]

**Pull request merged**